File: Source/MKL/examples/vmlc/source/vdunpackm.c
#include <stdio.h>
#include <math.h>
#include "func_interv.h"

#include "mkl_vml.h"

int main()
{
  double dA[VEC_LEN],dB1[VEC_LEN],dB2[VEC_LEN];
  int ma[VEC_LEN];

  int i=0,vec_len=VEC_LEN;

  for(i=0;i<vec_len;i++) {
    dA[i]=(double)i+1;
    dB1[i]=0.0;
    dB2[i]=0.0;
    ma[i]=i&1;
  }

  vdPackM(vec_len,dA,ma,dB1);
#if defined(MKL_VML_MIXED)
  vdUnpackM(vec_len,dB1,dB2,ma);
#elif defined(MKL_VML_LOWER)
  vdunpackm(&vec_len,dB1,dB2,ma);
#elif defined(MKL_VML_UPPER)
  VDUNPACKM(&vec_len,dB1,dB2,ma);
#else
# error You must define MKL_VML_CDECL for CDECL call convention or MKL_VML_STDCALL for STDCALL call convention.
#endif

  printf("vdUnpackM test/example programm\n\n");
  printf("           Before packing             After packing          After Unpacking\n");
  printf("===============================================================================\n");
  for(i=0;i<VEC_LEN;i++) {
    printf("% 25.14f ",dA[i]);
    printf("% 25.14f ",dB1[i]);
    printf("% 25.14f",dB2[i]);
    printf("\n");
  }
}

File: Source/MKL/examples/vmlf/source/vdunpackm.f
	programm MKL_VML_TEST

	include "func_interv.fi"

	real*8 dA(VEC_LEN)
	real*8 dB1(VEC_LEN)
	real*8 dB2(VEC_LEN)
	integer i
	integer ma(VEC_LEN)

	i=0

	do i=1,VEC_LEN
		dA(i)=i/1.0
		dB1(i)=0.0
		dB2(i)=0.0
		ma(i)=iand(i,1)
	end do

	call vdPackM(VEC_LEN,dA,ma,dB1)
	call VDUNPACKM(VEC_LEN,dB1,dB2,ma)

	print *,"vdUnpackM test/example programm"
	print *,""
	print *,""

	print *,"           Before packing             After packing          After Unpacking"
	print *,"==============================================================================="
	do i=1,VEC_LEN
		print 10,dA(i),dB1(i),dB2(i)
	end do

10	format (F25.13,F25.13,F25.13)

	end
